validation/resources.go

```diff
@@ -114,21 +137,35 @@ func validateResourceGroup(config ValConfig, rg types.Resource) error {
 	// 	return fmt.Errorf("error: invalid unit price (not positive fails)")
 	// }
 
-	return nil
+	return limits, nil
 }
 
-func validateResourceUnit(config ValConfig, unit types.Unit) error {
-	if unit.CPU > uint32(config.MaxUnitCPU) || unit.CPU < uint32(config.MinUnitCPU) {
-		return errors.Errorf("error: invalide unit cpu (%v > %v > %v fails)",
-			config.MaxUnitCPU, unit.CPU, config.MinUnitCPU)
+func validateResourceUnit(config ValConfig, units types.ResourceUnits) (resourceLimits, error) {
+	limits := newLimits()
+
+	if u := units.CPU; u != nil {
```

Comment:
do we ever ensure that `CPU` is declared?

Reply:
mean instantiation of big.Int value?

Reply:
Like did the user say the amount.  Basically, I think that `nil` should be an error

Reply:
i think we decided to keep them like this on initial, but i could be wrong.
where `CPU` and `Memory` should be `required`, storage though is more wide. for example deployment does not need storage (apart from docker image which in case of ML can occupy gigabytes) just default it to 0?

Reply:
This is validation, if cpu, memory, or storage are 0 or nil it is an error.

Reply:
I think I read your comment wrong because this line section allows nil on all of them.  Yes, I think defaulting storage to zero is okay for now.

Reply:
all resource values if they are != nil, validated against `ValConfig` values, if `Storage` is not nil and defaulted to 0, then check fails as default `MinUnitStorage` is 1024.

Reply:
Yes but if they are nil then it is an invalid resource

Reply:
yah, thats confusing, remember initially we though to parse resources into slice instead of using pointers and on validation step we woudln't deal with pointers

Reply:
I don't understand why we're talking past each other.  Validation should return an error when there is no cpu, memory.  If that is not the behavior then the behavior needs to be changed. 

Validation is used in multiple places.  It is the definition of a correctly-formed group/resource/etc.  It needs to be strict and it needs to be correct.